### PR TITLE
Removed `dataclass` usage for python3.6 compat

### PR DESCRIPTION
**Pull request by thekrampus**
See [this comment](https://github.com/rr-/docstring_parser/pull/19#issuecomment-620286669) on #19 by @EpicWink

### CODE REVIEW

File: docstring_parser/numpydoc.py [hunk 1]
```diff
@@ -57,8 +55,9 @@ class Section:
                 will be the first element of the ``args`` attribute list.
     """
 
-    title: str
-    key: str
+    def __init__(self, title: str, key: str):
```
Comment (rr-) [inline docstring_parser/numpydoc.py:58]: Let's put `-> None` here

Comment (robpkelly) [reply docstring_parser/numpydoc.py]: Right on. How about now?

Comment (thekrampus) [reply docstring_parser/numpydoc.py]: Right on. How about now?

**Comment by rr-** [comment]
Released as 0.7.1

File: docstring_parser/numpydoc.py (hunk 1 above)
Comment (rr-) [reply docstring_parser/numpydoc.py]: Thanks

**Pull request closed without merging**